spec/apis/event-listeners.spec.js

```diff
@@ -7,21 +7,27 @@ export function notStartedEventListeners() {
 			window.addEventListener("hashchange", () => {
 				if (window.location.hash === '#/a-new-hash')
 					hashchangeCalled = true;
+
+				checkTestComplete();
 			});
 			window.addEventListener("popstate", () => {
 				if (window.location.hash === '#/a-new-hash')
 					popstateCalled = true;
+
+				checkTestComplete();
 			});
 
 			window.location.hash = '#/a-new-hash';
-			setTimeout(() => {
-				expect(hashchangeCalled).toBe(true);
-				// https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/3740423/
-				if (isntIEOrEdge()) {
-					expect(popstateCalled).toBe(true);
+
+			function checkTestComplete() {
+				if (isIE()) {
+					// https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/3740423/
+					done(); // popstate isn't ever going to be called
+				} else if (hashchangeCalled && popstateCalled) {
+					// Wait for both hashchange and popstate events
+					done();
 				}
-				done();
-			}, 20);
```

Comment:
20 milliseconds is a race condition. The new way just has the test time out when jasmine gives up on it (10 seconds) if hashstate and popstate are not both called.